cmd/manager/collection/main.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/kabanero-io/kabanero-operator/pkg/controller/collection"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"github.com/operator-framework/operator-sdk/pkg/leader"
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	"github.com/operator-framework/operator-sdk/pkg/metrics"
+	"github.com/operator-framework/operator-sdk/pkg/restmapper"
+	sdkVersion "github.com/operator-framework/operator-sdk/version"
+	"github.com/spf13/pflag"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+
+	pipelinev1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+)
+
+// Change below variables to serve metrics on different host or port.
+var (
+	metricsHost       = "0.0.0.0"
+	metricsPort int32 = 8383
+)
+var log = logf.Log.WithName("collection cmd")
+
+// These variables are injected during the build using ldflags
+var GitTag string
+var GitCommit string
+var GitRepoSlug string
+var BuildDate string
+
+func printCollectionControllerData() {
+	log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+	log.Info(fmt.Sprintf("Version of operator-sdk: %v", sdkVersion.Version))
+
+	if len(GitTag) > 0 {
+		log.Info(fmt.Sprintf("kabanero-collection-operator Git tag: %s", GitTag))
+	}
+
+	if len(GitCommit) > 0 {
+		log.Info(fmt.Sprintf("kabanero-collection-operator Git commit: %s", GitCommit))
+	}
+
+	if len(GitRepoSlug) > 0 {
+		log.Info(fmt.Sprintf("kabanero-collection-operator Git repository: %s", GitRepoSlug))
+	}
+
+	if len(BuildDate) == 0 {
+		BuildDate = "unspecified"
+	}
+	log.Info(fmt.Sprintf("kabanero-collection-operator build date: %s", BuildDate))
+}
+
+func main() {
+	// Add the zap logger flag set to the CLI. The flag set must
+	// be added before calling pflag.Parse().
+	pflag.CommandLine.AddFlagSet(zap.FlagSet())
+
+	// Add flags registered by imported packages (e.g. glog and
+	// controller-runtime)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+
+	pflag.Parse()
+
+	// Use a zap logr.Logger implementation. If none of the zap
+	// flags are configured (or if the zap flag set is not being
+	// used), this defaults to a production zap logger.
+	//
+	// The logger instantiated here can be changed to any logger
+	// implementing the logr.Logger interface. This logger will
+	// be propagated through the whole operator, generating
+	// uniform and structured logs.
+	logf.SetLogger(zap.Logger())
+
+	printCollectionControllerData()
+
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		log.Error(err, "Failed to get watch namespace")
+		os.Exit(1)
+	}
+
+	// Get a config to talk to the apiserver
+	cfg, err := config.GetConfig()
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
+	ctx := context.TODO()
+
+	// Become the leader before proceeding
```

Comment:
This leader business is also a part of operator-sdk.  It's probably a good idea to enforce a singleton but we can cross that bridge later.